client/src/main/java/com/vesoft/nebula/encoder/NebulaCodecImpl.java

```diff
@@ -191,46 +191,73 @@ public SchemaProviderImpl genSchemaProvider(long ver, Schema schema) {
                                     type.type,
                                     len,
                                     nullable,
-                                    hasDefault ? null : null);
+                                    hasDefault ? col.getDefault_value() : null);
         }
         return schemaProvider;
     }
 
     /**
      * @param spaceName the space name
-     * @param schemaName the tag name or edge name
+     * @param tagName the tag name
      * @param names the property names
      * @param values the property values
      * @return the encode byte[]
      * @throws RuntimeException expection
      */
     @Override
-    public byte[] encode(String spaceName,
-                         String schemaName,
-                         List<String> names,
-                         List<Object> values)
-        throws RuntimeException {
-        if (names == null || values == null || names.size() != values.size()) {
-            throw new RuntimeException("NebulaCodeImpl input wrong value");
+    public byte[] encodeTag(String spaceName,
+                            String tagName,
+                            List<String> names,
+                            List<Object> values) throws RuntimeException  {
+        TagItem tag = metaCache.getTag(spaceName, tagName);
+        if (tag == null) {
+            throw new RuntimeException(
+                String.format("TagName `%s' is null", tagName));
```

Comment:
this msg may cause confuse, it is TagItem not tagName that is null.

Reply:
good catch. it's better to use the `TagItem is null when getting TagName `%s'`